feat(#2949): update checkbox to v2

bdfranck opened the pull request:
This PR updates the checkbox to match the [v2 Figma component](https://www.figma.com/design/Jpy1Ea5qglwnp1SgGnagY9/%E2%9D%96-Component-library-BETA?node-id=56969-469737&t=FCFdhIPcLhRaL1k3-1). Changes include:
- New SVGs for the checkmark and indeterminate dash
- Focus state uses an outline instead of a box shadow
- Compact state

twjeffery commented on libs/web-components/src/components/checkbox/Checkbox.svelte:
```diff
@@ -519,4 +552,44 @@ max-width: ${maxwidth};
+    outline-offset: var(--goa-space-3xs);
+    box-shadow: none;
+  }
+  .v2 .disabled:not(.error) .container {
```
**Bug: Disabled checked background uses wrong color**

This CSS applies `input-color-background-disabled` (greyscale-150, `#e9e9e9`) to ALL disabled checkboxes, overriding the correct `checkbox-color-bg-checked-disabled` (greyscale-300, `#b1b1b1`) for checked states.

**Current:** Disabled checked = `#e9e9e9`  (too light)
<img width="423" height="130" alt="Image" src="https://github.com/user-attachments/assets/c4f57f61-0272-40d7-9fa9-7cf6f512ac66" />

**Should be:** Disabled checked = `#b1b1b1`  (greyscale-300)
<img width="584" height="147" alt="Image" src="https://github.com/user-attachments/assets/5b31f8a6-9361-4beb-88c8-17ef60e3c19c" />

**Fix:** Only apply to unchecked disabled:

```suggestion
.v2 .disabled:not(.error):not(.selected) .container {
```

twjeffery commented on libs/web-components/src/components/checkbox/Checkbox.svelte:
```diff
@@ -519,4 +552,44 @@ max-width: ${maxwidth};
+  .v2 .error:not(.disabled) .container.selected:hover,
+  .v2 label:hover.error:not(.disabled) .container,
+  .v2 label:hover.error:not(.disabled) .container.selected {
+    background: var(--goa-input-color-background-error-hover);
```
**Bug: Error hover missing border change**

On hover, background changes (good!) but border stays the same (should also change).

**Two issues:**
1. Missing `checkbox-border-error-hover` token in design-tokens PR
2. Missing `box-shadow` change in this CSS

**Add `checkbox-border-error-hover` token to design-tokens PR**

```suggestion
  background: var(--goa-input-color-background-error-hover);
  box-shadow: var(--goa-checkbox-border-error-hover);
```

bdfranck commented:
I've amended my commit with the following changes:
- Reset and rebased branch against `dev`
- Added `size` property with a compact variant
- Addressed @twjeffery's CSS review comments

bdfranck commented on libs/web-components/src/components/checkbox/Checkbox.svelte:
```diff
@@ -84,6 +86,7 @@
     // hold on to the initial value to prevent losing it on check changes
     _value = value;
     _descriptionId = `description_${name}`;
+    mb ??= size === "compact" ? "s" : "m";
```
It looks like I'm the first person to use the [nullish coalescing assignment operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) (??=) in a Svelte component. Will this cause any issues? It appears to be widely supported. 🤔 

<img width="787" height="336" alt="Image" src="https://github.com/user-attachments/assets/05f372dc-2d23-4e4a-a810-a783a8dc689b" />

twjeffery reviewed (approved):
Looks great to me @bdfranck , thanks for making those changes.

tzuge commented:
:tada: This PR is included in version 1.10.0-dev.1 :tada:

The release is available on:
- [npm package (@dev dist-tag)](https://www.npmjs.com/package/@abgov/ui-components-common/v/1.10.0-dev.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/common-v1.10.0-dev.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

chrisolsen commented:
:tada: This PR is included in version 4.10.0-next.1 :tada:

The release is available on:
- [npm package (@next dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/4.10.0-next.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v4.10.0-next.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

chrisolsen commented:
:tada: This PR is included in version 1.40.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.40.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.40.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
